CONTRIBUTING.md

```diff
@@ -48,6 +48,58 @@ python run-clang-format.py \
 
 where `$CLANG_FORMAT` denotes the path to the downloaded binary.
 
+## Adding Third Party Libraries
+
+The following steps outline how to add third party libraries to torchtext:
+
+1. Add the git repo you care about as a submodule. Here is a great
+   [tutorial](https://www.atlassian.com/git/tutorials/git-submodule) on working with submodules in git.
+   - Navigate to `third_party/` folder and run `git submodule add <repo-URL>`
+   - Verify the newly added module is present in the
+     [`.gitmodules`](https://github.com/pytorch/text/blob/main/.gitmodules) file
+2. Update
+   [`third_party/CMakeLists.txt`](https://github.com/pytorch/text/blob/70fc1040ee40faf129604557107cc59fd51c4fe2/third_party/CMakeLists.txt#L8)
+   to add the following line: `add_subdirectory(<name-of-submodule-folder> EXCLUDE_FROM_ALL)`
+3. (Optional) If any of the files within the `csrc/` folder make use of the newly added third party library then
+   - Add the new submodule folder to
+     [`​​LIBTORCHTEXT_INCLUDE_DIRS`](https://github.com/pytorch/text/blob/70fc1040ee40faf129604557107cc59fd51c4fe2/torchtext/csrc/CMakeLists.txt#L24)
+     and to
+     [`EXTENSION_INCLUDE_DIRS`](https://github.com/pytorch/text/blob/70fc1040ee40faf129604557107cc59fd51c4fe2/torchtext/csrc/CMakeLists.txt#L119)
+   - Add the submodule name to
```

Comment:
Technically, its "target" names, defined by the third party CMakeLists.

https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#introduction

Reply:
Updated